Scripts/Editor/UI/EditorInputDialog.cs

```diff
@@ -0,0 +1,178 @@
+
+            EditorInputDialog window = CreateInstance<EditorInputDialog>();
+            window.Init(
+                maxPos,
+                title,
+                description,
+                inputTextLabel,
+                okBtnText,
+                cancelBtnText,
+                defaultInputText);
+            window.OnComplete += (resultString) => result = resultString;
+            window.ShowModal();
+
+            return result;
+        }
+
+
+        private event Action<string> OnComplete;
+
+        private string m_Description;
+        private string m_InputTextLabel;
+        private string m_OKBtnText;
+        private string m_CancelBtnText;
+        private Vector2 m_MaxScreenPosition;
+
+        private bool m_HasInitializedPosition;
+        private string m_InputText;
+
+        private void Init(
+            Vector2 maxPos,
+            string titleText,
+            string description,
+            string inputTextLabel,
+            string okBtnText,
+            string cancelBtnText,
+            string defaultInputText = null)
+        {
+            m_MaxScreenPosition = maxPos;
+            titleContent = new GUIContent(titleText);
+            m_Description = description;
+            m_InputTextLabel = inputTextLabel;
+            m_OKBtnText = okBtnText;
+            m_CancelBtnText = cancelBtnText;
+            m_InputText = defaultInputText ?? string.Empty;
+        }
+
+        private void OnGUI()
+        {
+            Event currentEvent = Event.current;
+            CheckKeyboard(currentEvent);
+
+            Rect rect = EditorGUILayout.BeginVertical();
+
+            EditorGUILayout.Space(12);
+            EditorGUILayout.LabelField(m_Description);
+
+            EditorGUILayout.Space(8);
+            GUI.SetNextControlName(CONTROL_NAME_INPUT_TEXT);
+            m_InputText = EditorGUILayout.TextField(m_InputTextLabel, m_InputText);
+            GUI.FocusControl(CONTROL_NAME_INPUT_TEXT);
+            EditorGUILayout.Space(12);
+
+            Rect controlRect = EditorGUILayout.GetControlRect();
+            controlRect.width *= 0.5f;
+            if (GUI.Button(controlRect, m_OKBtnText))
+            {
+                Complete(false);
+            }
+
+            controlRect.x += controlRect.width;
+            if (GUI.Button(controlRect, m_CancelBtnText))
+            {
+                Complete(true);
+            }
+
+            EditorGUILayout.Space(8);
+            EditorGUILayout.EndVertical();
+
+            if (rect.width != 0 && minSize != rect.size)
+            {
+                minSize = maxSize = rect.size;
+            }
+
+            InitializePosition(currentEvent);
+        }
+
+        private void CheckKeyboard(Event currentEvent)
+        {
+            if (currentEvent.type != EventType.KeyDown)
+            {
+                return;
+            }
+            
+            // ReSharper disable once SwitchStatementMissingSomeEnumCasesNoDefault
```

Comment:
you could probably just add a `default` condition to the switch that called `break` and you wouldn't need this.

Reply:
I tried and it just stays greyed out and I get a different ReSharper error to disable once instead.